packages/qwik/src/core/container/serializers.ts

```diff
@@ -193,7 +194,12 @@ const URLSerializer = /*#__PURE__*/ serializer<URL>({
   $prefix$: '\u0005',
   $test$: (v) => v instanceof URL,
   $serialize$: (obj) => obj.href,
-  $prepare$: (data) => new URL(data),
+  $prepare$: (data) => {
+    if (isBrowser && !isDev) {
+      return new URL(data + document.location.search);
```

Comment:
I don't think this is the solution, now you're adding the location to all URLs.

Furthermore, containers only get serialized on the server, so this code doesn't do anything.

Can you add a test first and then try to fix it?